### PR TITLE
WPCOM: Remove `undocumented().me()` completely

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This PR migrates the last `wpcom.undocumented()` method - `getReceipt()` to `wpcom.req`. 

We also remove the unnecessary `undocumented().me()`, while removing its last usages, that were just proxying requests to the `wpcom.js` library.

Part of the ongoing effort to get rid of `wpcom.undocumented()` and part of #58458.

~Depends on #58452 and has to be rebased once it gets merged.~

#### Testing instructions

* Purchase a plan and verify the thank you page still works well and displays purchase details correctly.
* Navigate to `/me/purchases/billing` and click on a receipt. Verify the page still works well.
* Verify you can still change and save settings in Me -> Account Settings correctly.
* Verify you can still change your 2FA SMS settings (namely, the phone number).
* Verify you can still change your email address through the domain contact info management page.
* Verify all tests still pass.

### CODE REVIEW

File: client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx [hunk 1]
```diff
@@ -283,8 +281,7 @@ class EditContactInfoFormCard extends Component {
 
 		const { email } = newContactDetails;
 		if ( updateWpcomEmail && email && this.props.currentUser.email !== email ) {
-			wpcom
-				.me()
+			wp.me()
```
Comment (jsnajdr) [inline client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx:284]: Interesting, the `wpcom.undocumented().me()` was inherited from `wpcom.me()`, and included all the methods... I was confused at first how the `.settings()` method could work at all.

**Review by jsnajdr — APPROVED** [review]
Nice milestone, getting rid of `wpcom.undocumented().me()` 🙂

File: client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx (hunk 1 above)
Comment (tyxla) [reply client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx:284]: That part was pretty confusing, so ambiguous! You could use `wpcom.undocumented().me().settings()` or `wpcom.me().settings()` and they would both do the same thing! And having `.settings()` available for `wpcom.undocumented()` itself doesn't make it less confusing 😅

**Comment by tyxla** [comment]
> Nice milestone, getting rid of `wpcom.undocumented().me()` 🙂

Thanks! Would be nice to get a review of #58452 too, because it's a pre-requisite of this one.

**Pull request merged**